### PR TITLE
backporting Migration guide

### PR DESCRIPTION
**Pull request by zbendhiba**

### CODE REVIEW

File: docs/modules/ROOT/pages/migration-guide/3.0.0.adoc [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+| camel-quarkus-activemq               | camel-quarkus-jms, camel-quarkus-sjms, camel-quarkus-amqp
+| camel-quarkus-caffeine-lrucache      | camel-quarkus-ignite, camel-quarkus-infinispan
+| camel-quarkus-elasticsearch-rest     | TODO Coming camel-quarkus-elasticsearch
+| camel-quarkus-johnzon                | camel-quarkus-jackson, camel-quarkus-fastjson, camel-quarkus-gson
+| camel-quarkus-microprofile-metrics   | camel-quarkus-micrometer, camel-quarkus-opentelemtry
+| camel-quarkus-opentracing            | camel-quarkus-micrometer, camel-quarkus-opentelemtry
+| camel-quarkus-optaplanner            | none
+| camel-quarkus-rabbitmq               | none
+| camel-quarkus-vm                     | camel-quarkus-seda
+| camel-quarkus-xstream                | camel-quarkus-jacksonxml
```
Comment (ppalaga) [inline docs/modules/ROOT/pages/migration-guide/3.0.0.adoc:29]: There is more: 
Atmos - removed from Camel
Avro RPC https://github.com/apache/camel-quarkus/issues/4462
DataSonnet - should come back https://github.com/apache/camel-quarkus/issues/4685
Dozer - missing in Camel 4, altough https://camel.apache.org/manual/camel-4-migration-guide.html does not say so
HBase - removed from Camel
Iota - removed from Camel
JBPM  - removed from Camel
JClouds  - removed from Camel
Milo - removed from Camel
smallrye-reactive-messaging - will come back https://github.com/apache/camel-quarkus/issues/4637
Tika - should come back https://github.com/apache/camel-quarkus/issues/4470
XML Security Sign - will come back https://github.com/apache/camel-quarkus/issues/4684

**Review by ppalaga — COMMENTED** [review]
Thanks @zbendhiba for taking care! by comparing the boms from 2.13 and main, I have found that there are a few more removed extensions. I have added issues to re-add for some of those. Please see inline.

File: docs/modules/ROOT/pages/migration-guide/3.0.0.adoc (hunk 1 above)
Comment (ppalaga) [reply docs/modules/ROOT/pages/migration-guide/3.0.0.adoc:29]: Sorry, now I noticed this is a bakcport. The same applies to `main`

Comment (zbendhiba) [reply docs/modules/ROOT/pages/migration-guide/3.0.0.adoc:29]: Ok then let me add those to main, and port back here

Comment (aldettinger) [reply docs/modules/ROOT/pages/migration-guide/3.0.0.adoc:29]: The alternative extensions column is great :+1:

**Comment by zbendhiba** [comment]
I think the failure is unrelated, I'll merge this to have a link for blog post

**Pull request merged**